src/questions/chapter5.yaml

```diff
@@ -0,0 +1,1075 @@
+      key: B
+      order: 1
+    - content: An instance of an abstract cannot be created.
+      key: C
+      order: 2
+    - content: All of these
+      key: D
+      order: 3
+
+- uuid: 41899c0d-e985-42b6-b532-34760b1d1d2b
+  type: SingleChoice
+  content:  If the derived contract of an abstract contract is also not implementing the incomplete functions then that derived contract will also be marked as
+  hint: noHint
+  explanation: 
+  answerKeys:
+    - B
+  topic: Abstraction
+  subTopics:
+    - abstract
+  difficultyLevel: Low
+  choices:
+    - content: interface
+      key: A
+      order: 0
+    - content: abstract
+      key: B
+      order: 1
+    - content: library
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 67c5c94c-42f8-4c65-a2dc-a7cdc121db6a
+  type: SingleChoice
+  content: Which of the following keyword is used to declare an abstract contract?
+  hint: noHint
+  explanation: An abstract contract is declared using the abstract keyword. 
+  answerKeys:
+    - C
+  topic: Abstraction
+  subTopics:
+    - abstract
+  difficultyLevel: Low
+  choices:
+    - content: interface
+      key: A
+      order: 0
+    - content: library
+      key: B
+      order: 1
+    - content: abstract
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 4d443bb2-d84b-475e-8337-55ab3fcb4d33
+  type: SingleChoice
+  content: Which of the below statement is correct regarding constructor in solidity?
+  hint: noHint
+  explanation: Solidity provides a constructor declaration inside the smart contract and it invokes only once when the contract is deployed and is used to initialize the contract state.
+  answerKeys:
+    - C
+  topic: Constructor
+  subTopics:
+    - constructor
+  difficultyLevel: Medium
+  choices:
+    - content: A constructor can be called multiple times.
+      key: A
+      order: 0
+    - content: A constructor can not accept arguments.
+      key: B
+      order: 1
+    - content: A constructor can be invoked only once.
+      key: C
+      order: 2
+    - content: None
+      key: D
+      order: 3
+
+- uuid: 60ad4e34-501c-4039-a3bb-da0a5730a52b
+  type: SingleChoice
+  content: Which of the following key word is used to declare a constructor?
+  hint: noHint
+  explanation: A Constructor is defined using a constructor keyword without any function name followed by an access modifier.
+  answerKeys:
+    - D
+  topic: Constructor
+  subTopics:
+    - constructor
+  difficultyLevel: Low
+  choices:
+    - content: pragma
+      key: A
+      order: 0
+    - content: abstract
+      key: B
+      order: 1
+    - content: interface
+      key: C
+      order: 2
+    - content: constructor
+      key: D
+      order: 3
+
+- uuid: 8e9bf2dd-0481-4edf-88db-5dfad785f513
+  type: SingleChoice
+  content: Choose the correct statement from below options about constructor in solidity. 
```

Comment:
Bothe these statements are not related to one another 
`A parameter value can be defined at the run time in a constructor`

and 
`A constructor can also restrict method call.`